src/mlpack/methods/ann/layer/residual.hpp

```diff
@@ -0,0 +1,127 @@
+/**
+ * @file methods/ann/layer/residual.hpp
+ * @author Shubham Agrawal
+ *
+ * Base class for neural network layers that are wrappers around other layers.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RESIDUAL_HPP
+#define MLPACK_METHODS_ANN_LAYER_RESIDUAL_HPP
+
+#include "../make_alias.hpp"
+#include "multi_layer.hpp"
+
+namespace mlpack {
+namespace ann {
+
+/**
+ * A lambda "map-reduce" is a layer that is a wrapper around other layers.  
+ * It passes the input through all of its child layers sequentially, returning
+ * the output from reducing the output.
+ *
+ * @tparam MatType Matrix representation to accept as input and use for
+ *    computation.
+ */
+template<typename MatType>
+class ResidualType : public MultiLayer<MatType>
```

Comment:
Are you sure that `Residual` is the right name here?  Not `Reduction`?

Reply:
If you're hardcoding the add reduction here, perhaps a better name is `AddMerge` (that is the name of another old layer), but I'm not too picky.  In any case though it does seem like "residual" is not the right word, unless this is a residual in some sense I'm not familiar with.